File: Library/Models/Music.cs
using Library.Enums;

namespace Library.Models;

public class Music
{
    protected const string splitCharacter = "/";

    protected string? id;
    protected string? vietnameseName, simplifiedChineseName, traditionalChineseName, pinyinName;
    protected string? vietnameseDescription, simplifiedChineseDescription, traditionalChineseDescription;

    public string? Id => id;

    public Music SetName(Language language, string name)
    {
        switch (language)
        {
            case Language.Vietnamese: vietnameseName = name; break;
            case Language.SimplifiedChinese: simplifiedChineseName = name; break;
            case Language.TraditionalChinese: traditionalChineseName = name; break;
            case Language.Pinyin: pinyinName = name; break;
        }    
        return this;
    }

    public Music SetDescription(Language language, string description)
    {
        switch (language)
        {
            case Language.Vietnamese: vietnameseDescription = description; break;
            case Language.SimplifiedChinese: simplifiedChineseDescription = description; break;
            case Language.TraditionalChinese: traditionalChineseDescription = description; break;
        }
        return this;
    }

    public string? GetName(Language language)
    {
        return language switch
        {
            Language.SimplifiedChinese => simplifiedChineseName,
            Language.TraditionalChinese => traditionalChineseName,
            Language.Pinyin => pinyinName,
            Language.Vietnamese => vietnameseName,
            _ => throw new NotImplementedException(),
        };
    }

    public string? GetDescription(Language language)
    {
        return language switch
        {
            Language.SimplifiedChinese => simplifiedChineseDescription,
            Language.TraditionalChinese => traditionalChineseDescription,
            _ => vietnameseDescription
        }; 
    }

    public string GetPhotoUrl(PhotoQuality photoQuality)
    {
        return photoQuality switch
        {
            PhotoQuality.Small or PhotoQuality.Medium or PhotoQuality.Large
                => $"https://y.qq.com/music/photo_new/T001R{ (int)photoQuality }x{ (int)photoQuality }M000{ id }.jpg",
            _ => $"https://i.ytimg.com/vi/{ id }/{ photoQuality.ToString().ToLower() }.jpg"
        };
    }
}

File: Music.Android/television/app/src/main/java/com/zitherharpmusic/zhmtelevision/music/Music.java
package com.zitherharpmusic.zhmtelevision.music;

import org.jetbrains.annotations.NotNull;

public class Music {
    public static final String EMPTY_CHARACTER = "";
    public static final String SPACE_CHARACTER = " ";
    public static final String SPLIT_CHARACTER = "/";
    public static final String ENTER_CHARACTER = "\n";

    protected String id;
    protected String vietnameseName;
    protected String pinyinName;
    protected String simplifiedChineseName;
    protected String traditionalChineseName;
    protected String vietnameseDescription;
    protected String simplifiedChineseDescription;
    protected String traditionalChineseDescription;

    public String getId() {
        return id;
    }

    public String getName(@NotNull Language language) {
        switch (language) {
            case PINYIN:
                return pinyinName;
            case SIMPLIFIED_CHINESE:
                return simplifiedChineseName;
            case TRADITIONAL_CHINESE:
                return traditionalChineseName;
            case VIETNAMESE:
                return vietnameseName;
        }
        throw new NullPointerException();
    }

    public String getDescription(@NotNull Language language) {
        switch (language) {
            case PINYIN:
            case SIMPLIFIED_CHINESE:
                return simplifiedChineseDescription;
            case TRADITIONAL_CHINESE:
                return traditionalChineseDescription;
            case VIETNAMESE:
                return vietnameseDescription;
        }
        throw new NullPointerException();
    }

    public Music setName(@NotNull Language language, String name) {
        switch (language) {
            case VIETNAMESE:
                this.vietnameseName = name;
                return this;
            case PINYIN:
                this.pinyinName = name;
                return this;
            case SIMPLIFIED_CHINESE:
                this.simplifiedChineseName = name;
                return this;
            case TRADITIONAL_CHINESE:
                this.traditionalChineseName = name;
                return this;
        }
        return this;
    }

    public Music setDescription(@NotNull Language language, String description) {
        switch (language) {
            case VIETNAMESE:
                this.vietnameseDescription = description;
                return this;
            case SIMPLIFIED_CHINESE:
                this.simplifiedChineseDescription = description;
                return this;
            case TRADITIONAL_CHINESE:
                this.traditionalChineseDescription = description;
                return this;
        }
        return this;
    }

    public String getPhotoUrl(PhotoQuality photoQuality) {
        if (photoQuality == PhotoQuality.SMALL || photoQuality == PhotoQuality.MEDIUM || photoQuality == PhotoQuality.LARGE) {
            return String.format("https://y.qq.com/music/photo_new/T001R%sM000%s.jpg", photoQuality.getSize(), id);
        } else {
            return String.format("https://i.ytimg.com/vi/%s/%s.jpg", id, photoQuality.name().toLowerCase());
        }
    }

    public String getShareUrl() {
        return "https://youtu.be/" + id;
    }

    public String toString(Language language) {
        return id + ": " + getName(language);
    }
}
